src/legacy/server/saved_objects/service/lib/included_fields.ts

```diff
@@ -17,26 +17,30 @@
  * under the License.
  */
 
+function toArray(value: string | string[]): string[] {
+  return typeof value === 'string' ? [value] : value;
+}
 /**
  * Provides an array of paths for ES source filtering
- *
- * @param {string} type
- * @param {string|array} fields
- * @returns {array}
  */
-export function includedFields(type, fields) {
-  if (!fields || fields.length === 0) return;
+export function includedFields(type: string | string[] = '*', fields?: string[] | string) {
+  if (!fields || fields.length === 0) {
+    return;
+  }
 
   // convert to an array
-  const sourceFields = typeof fields === 'string' ? [fields] : fields;
-  const sourceType = type || '*';
+  const sourceFields = toArray(fields);
+  const sourceType = toArray(type);
 
-  return sourceFields
-    .map(f => `${sourceType}.${f}`)
+  return sourceType
+    .map(t => sourceFields.map(f => `${t}.${f}`))
```

Comment:
The `map` and `reduce` can be combined if you want to avoid an extra iteration:

```ts
return sourceType
  .reduce((acc, t) => [...acc, ...sourceFields.map(f => `${t}.${f}`)], [])
```

Reply:
:+1: What I actually want is`flatMap` but not supported yet :(